File: 2053.py
class Solution:
    def kthDistinct(self, arr: List[str], k: int) -> str:
        d = collections.Counter(arr)
        i = 0
        for w in arr:
            if d[w] == 1:
                i += 1
            if i == k:
                return w
            """
            The loop body is faster than
            if d[w] == 1:
                i += 1
                if i == k:
                    return w
            """
        return ""

File: 2053.cpp
class Solution {
public:
    string kthDistinct(vector<string>& arr, int k) {
        unordered_map<string, int> m;
        for (string& s : arr) {
            m[s]++;
        }
        for (string& s : arr) {
            if (m[s] == 1) {
                k--;
                if (k == 0) {
                    return s;
                }
            }
        }
        return "";
    }
};
